### PR TITLE
HDDS-1830 OzoneManagerDoubleBuffer#stop should wait for daemon thread to die

### PR DESCRIPTION
**Pull request by smengcl**

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java [hunk 1]
```diff
@@ -64,7 +65,7 @@
   private final OMMetadataManager omMetadataManager;
   private final AtomicLong flushedTransactionCount = new AtomicLong(0);
   private final AtomicLong flushIterations = new AtomicLong(0);
-  private volatile boolean isRunning;
+  private AtomicBoolean isRunning = new AtomicBoolean(true);
```
Comment (arp7) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: Let's also make it final.

**Comment by smengcl** [comment]
Oops. Thanks for pointing it out @arp7 . Just updated the commit.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java [hunk 2]
```diff
@@ -64,7 +65,7 @@
   private final OMMetadataManager omMetadataManager;
   private final AtomicLong flushedTransactionCount = new AtomicLong(0);
   private final AtomicLong flushIterations = new AtomicLong(0);
-  private volatile boolean isRunning;
+  private final AtomicBoolean isRunning = new AtomicBoolean(true);
```
Comment (bharatviswa504) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: IsRunning to true should be set only after deamon.start() right?

Comment (smengcl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java]: Thanks for pointing this out. Resolved.

**Comment by arp7** [comment]
I merged this before CI had a chance to complete. Sorry about that. Let me revert and retest.

**Comment by arp7** [comment]
I am not sure how revert will work with the PR. Will leave it alone for now. Let's see if subsequent CI runs uncover any issue with this change.

**Pull request merged**